src/_pytest/junitxml.py

```diff
@@ -110,20 +110,14 @@ def record_testreport(self, testreport):
         classnames = names[:-1]
         if self.xml.prefix:
             classnames.insert(0, self.xml.prefix)
-        attrs = {
-            "classname": ".".join(classnames),
-            "name": bin_xml_escape(names[-1]),
```

Comment:
Unfortunately this one iis a breaking change, well have to sort out the details

Reply:
Please elaborate. What does it break?

Reply:
@jhunkeler off hand we cant rule out that people rely on the invalid extra attributes we output

Reply:
or to elaborate - junitxml output has been brokenly incorrect for so long that we now are likely to break other peoples code with the fix and we have to account for that in some way (be it a major release for this fix or a opt-in for correct behaviour that will later be a opt out and finally be phased out

Reply:
How about creating an `pytest.ini` option to configure this then? `junitxml_strict=true`?

Reply:
And this would need to be based on `features` btw.